### DISCUSSION TITLE
Average of numbers

### DISCUSSION THREAD

**Discussion by DeveloperPeach** [TheAlgorithms/Python#7981] [Q&A]
What is the code used to find the average/mean of any amount of numbers?

**Comment by richard-hajek** [comment]
Use this https://github.com/TheAlgorithms/Python/blob/master/maths/average_mean.py for a small amount of numbers (<100000). For higher amount of numbers you may have to use cumulative running mean and use the last value.

**Reply by richard-hajek** [reply]
See https://stackoverflow.com/a/23493727 for example on cumulative rolling mean

**Comment by Tanya-Kanaujia** [comment]
> What is the code used to find the average/mean of any amount of numbers?

ADD the numbers and divide by 2

**Reply by richard-hajek** [reply]
[4, 8, 6]

Add numbers div 2 = 9

not average

**Comment by Have798** [comment]
To find the average of any amount of numbers, you can use the following Python code:


def calculate_average(numbers):
    total = 0
    count = 0
    for num in numbers:
        total += num
        count += 1
    return total / count


This code defines a function called calculate_average that takes a list of numbers as input. It initializes two variables, total and count, to 0. Then, it iterates over each number in the list and adds it to the total variable while incrementing the count variable. Finally, it returns the average by dividing the total by the count.

To use this function, you can pass a list of numbers as an argument:

numbers = [2, 4, 6, 8, 10]
average = calculate_average(numbers)
print(average)  # Output: 6.0


In this example, the list [2, 4, 6, 8, 10] is passed to the calculate_average function, and it returns the average, which is 6.0.
